core/executor/src/allocator.rs

```diff
@@ -78,35 +115,45 @@ impl FreeingBumpHeapAllocator {
 	}
 
 	/// Gets requested number of bytes to allocate and returns a pointer.
+	/// The minimum size which can be allocated is 8 bytes.
 	/// The maximum size which can be allocated at once is 16 MiB.
 	pub fn allocate(&mut self, size: u32) -> Result<u32> {
 		if size > MAX_POSSIBLE_ALLOCATION {
 			return Err(Error::RequestedAllocationTooLarge);
 		}
 
-		let size = size.max(8);
+		let size = size.max(MIN_POSSIBLE_ALLOCATION);
 		let item_size = size.next_power_of_two();
-		if item_size + 8 + self.total_size > self.max_heap_size {
+		if item_size + PREFIX_SIZE + self.total_size > self.max_heap_size {
 			return Err(Error::AllocatorOutOfSpace);
 		}
 
 		let list_index = (item_size.trailing_zeros() - 3) as usize;
 		let ptr: u32 = if self.heads[list_index] != 0 {
 			// Something from the free list
 			let item = self.heads[list_index];
+			let ptr = item + PREFIX_SIZE;
+			if ptr + item_size > self.max_heap_size {
+				return Err(Error::AllocatorOutOfSpace);
```

Comment:
This should not be possible given that the free list entry is looked up by the item size. Perhaps this should be an assertion rather than returning an error?

Reply:
Yeah good point, I had also thought about an assertion when writing the code, but thought it might be bad to panic here. After talking to Basti I've now changed it to an assertion.